go/adbc/ext.go

```diff
@@ -78,3 +81,55 @@ type OTelTracing interface {
 	// Gets the initial span attributes for any newly started span.
 	GetInitialSpanAttributes() []attribute.KeyValue
 }
+
+// IngestStreamOption bundles the most-common IngestStream settings.
+// Any other ADBC options can go into Extra.
+type IngestStreamOption struct {
+	TargetTable string            // required
+	IngestMode  string            // required, e.g. adbc.OptionValueIngestModeCreateAppend, or OptionValueIngestModeReplace
+	Extra       map[string]string // any other stmt.SetOption(...) args
+}
+
+// IngestStream is a helper for executing a bulk ingestion. This is a wrapper around
+// the five-step boilerplate of NewStatement, SetOption, Bind,
+// Execute, and Close.
+//
+// This is not part of the ADBC API specification.
+func IngestStream(ctx context.Context, cnxn Connection, reader array.RecordReader, opt IngestStreamOption) (int64, error) {
+	// 1) Create a new statement
+	stmt, err := cnxn.NewStatement()
+	if err != nil {
+		return -1, fmt.Errorf("IngestStream: NewStatement: %w", err)
+	}
+	defer func() {
+		err = errors.Join(err, stmt.Close())
+	}()
+
+	// 2) Bind the record batch stream
+	if err = stmt.BindStream(ctx, reader); err != nil {
+		return -1, fmt.Errorf("IngestStream: BindStream: %w", err)
+	}
+
+	// Set required options
+	if err := stmt.SetOption(OptionKeyIngestTargetTable, opt.TargetTable); err != nil {
```

Comment:
remove the `:`, you're shadowing the `err` from before so the `errors.Join` isn't going to work this way.